commons/src/main/java/com/powsybl/commons/datasource/DirectoryDataSource.java

```diff
@@ -59,7 +62,8 @@ protected OutputStream getCompressedOutputStream(OutputStream os) throws IOExcep
 
     private Path getPath(String fileName) {
         Objects.requireNonNull(fileName);
-        return directory.resolve(fileName + getCompressionExt());
+        FileInformation fileInformation = new FileInformation(fileName, false);
+        return directory.resolve(fileInformation.getCompressionFormat() == null ? fileName + getCompressionExtension() : fileName);
```

Comment:
mention this fix/change or put in a separate PR ?

This changes compressed directory data source to allow newinputstream newoutputstream and exists to work with "foo.xiidm" when the file is "foo.xiidm.gz"

so it introduces the concept of "filenameOrUncompressedFilename" ?

Reply:
added info about this in the PR comment + in the breaking change section